metadata/src/main/java/org/apache/kafka/image/publisher/MetadataPublisher.java

```diff
@@ -40,33 +40,30 @@ public interface MetadataPublisher extends AutoCloseable {
     String name();
 
     /**
-     * Publish a new cluster metadata snapshot that we loaded.
+     * Handle a change in the current controller.
      *
-     * @param delta    The delta between the previous state and the new one.
-     * @param newImage The complete new state.
-     * @param manifest The contents of what was published.
+     * @param newLeaderAndEpoch The new quorum leader and epoch. The new leader will be
+     *                          OptionalInt.empty if there is currently no active controller.
      */
-    void publishSnapshot(
-            MetadataDelta delta,
-            MetadataImage newImage,
-            SnapshotManifest manifest
-    );
+    default void handleControllerChange(LeaderAndEpoch newLeaderAndEpoch) { }
```

Comment:
nit: "publishControllerChange" to keep the naming consistent?

Reply:
How about
`void onControllerChange(LeaderAndEpoch newLeaderAndEpoch)`
and
`void onMetadataUpdate(MetadataDelta delta, MetadataImage newImage, LoaderManifest manifest);`?
Might be clearer.